spec/controllers/submissions_controller_spec.rb

```diff
@@ -184,6 +184,59 @@
           expect(files['TestShapes.java']).to be_nil
         end
       end
+
+      context 'when creating a folder with required files' do
+        before :each do
+          @assignment.update(
+            only_required_files: true,
+            assignment_files_attributes: [{ filename: 'test_zip/zip_subdir/TestShapes.java' }]
+          )
+        end
+        it 'should upload a folder named test_zip' do
+          post_as @student, :update_files,
+                  params: { course_id: course.id, assignment_id: @assignment.id,
+                            new_folders: ['test_zip'] }
+          expect(response).to have_http_status :ok
+        end
+        it 'should upload a subdirectory' do
+          post_as @student, :update_files,
+                  params: { course_id: course.id, assignment_id: @assignment.id,
+                            new_folders: ['test_zip/zip_subdir'] }
```

Comment:
shouldn't the params be `path: 'test_zip', new_folders: ['zip_subdir']`?

and the same for the test at line 213 below

Reply:
I don't think so.  `update_files` does take a path parameter. it's used when creating new files, but it is not used in creating new folders (see `handleCreateFolder` in `submission_file_manager.jsx`). 

Instead just the grouping_id and the list of new folders (with full relative paths) are sent to update_files. I'd actually like to discuss that briefly with you - I think I understand why, but I want to confirm

Reply:
I see... you're right. It probably doesn't matter but I think I expect it to be the other way just so that it's consistent with the file upload bit. I guess the function handles it either way so let's leave it like this for now since that mimics what is being sent by the front end.